MovingBlocks/Terasology
# Add memory for confirmed trusted URLs to let the game open in a browser

Closes #3414.

![image](https://user-images.githubusercontent.com/26657147/46326907-5fc12700-c5d5-11e8-96d0-87073b414f66.png)


### Contains
1. Add a new checkbox on the confirmation dialog.

### How to test
1. Add the button widget to some existing screen and set up a link. For example, you can change the `updateOpenInBrowserButton()` in `ModuleDetailsScreen.java` to the folowing:
```java
private void updateOpenInBrowserButton() {
        openInBrowser.setEnabled(false);
        openInBrowser.setLink("https://github.com/llvieira")
            .setNuiManager(getManager())
            .setTranslationSystem(translationSystem)
}
``` 
2. Go the the screen where you enabled the open-in-browser functionality.
2. Mark the checkbox to save the link. All the trusted URLs will be stored in `config.cfg`.
```shell
"webBrowserConfig": {
    "trustedUrls": [
      "https://github.com/llvieira"
    ]
  }
```
3. Close the game, open it again, go to the same screen.
4. Check that the default web browser open the link without asking for permission.
5. Try the same without making the checkbox. The popup will always appear.

### TODO - WIP part
1. Just save the root domain instead of the entirely URL. **[GCI task]**
